### ISSUE TITLE
Controlled TextInput component breaks Pinyin autocompletion in v0.26.0

### ISSUE THREAD

**Issue by bnham** [facebook/react-native#8265]
There is a regression in React Native v0.26.0. Specifically, a simple controlled TextInput component (both single and multiline) like this:

```
var ControlledTextInput = React.createClass({
  getInitialState: function() {
    return {value: ''};
  },
  _onChangeText: function(text) {
    this.setState({value: text});
  },
  render: function() {
    return <TextInput
      value={this.state.value}
      style={styles.default}
      onChangeText={this._onChangeText.bind(this)}
      {...this.props}
      />;
  }
});
```

breaks the Chinese Pinyin keyboard's autocompletion feature. For instance, after typing the "n" here, the keyboard should offer an autocompletion to "你", but instead the keyboard offers an autocompletion to ",":

![pinyin_autocompletion](https://cloud.githubusercontent.com/assets/3528856/16213933/2857d448-3708-11e6-8428-76bd1de2675d.png)

I bisected and traced this regression down to 9be37ca, which is the commit that updated us to using react 15.0.2-alpha.2.

The reason for the breakage is that after the problematic diff, `-[RCTTextView setText:]` gets called twice for every keypress: once with the old text and once with the new text with the new character appended. By calling `setText:` with the old text, we end up calling `-[UITextView setText:]`, which clears the marked autocompletion state of the text view and breaks multi-character autocomplete keyboards like Chinese Pinyin.

See https://github.com/facebook/react-native/pull/8140 for a PR that tries to work around this issue. I'm going to close that PR for now and work on trying to fix the regression rather than applying a workaround.

**Comment by wasabia** [comment]
will be fixed in v0.28.0?

**Comment by bnham** [comment]
The intention is to fix it as soon as we can, yes.

**Comment by bnham** [comment]
This commit in react seems to be the root cause of the duplicate `-setText:` calls: https://github.com/facebook/react/pull/6572

**Comment by bnham** [comment]
It turns out this bug was fixed by another diff: 26aa27da630a8875dfe64f616b6aca59eaa89aaf. This diff is included in v0.28.0 so this bug is fixed.
